lib/js/index.js

```diff
@@ -19,6 +19,29 @@ exports.privateKeyExport = function (privateKey, compressed) {
   return g.mul(d).toPublicKey(compressed)
 }
 
+exports.privateKeyNegate = function (privateKey) {
+  var bn = BN.fromBuffer(privateKey)
+
+  if (bn.isZero()) {
+    return Buffer.alloc(32)
+  }
+
+  if (bn.ucmp(BN.n) !== 0) {
+    while (bn.isOverflow()) {
+      bn.isub(BN.n)
+    }
+  }
```

Comment:
Why we need `while` loop here? Loop body will be executed 1 time at max..
```js
exports.privateKeyNegate = function (privateKey) {
  var bn = BN.fromBuffer(privateKey)
  if (bn.isZero()) return Buffer.alloc(32)

  if (bn.ucmp(BN.n) > 0) bn.isub(BN.n)
  return BN.n.sub(bn).toBuffer()
}
```

Reply:
You are right, the `while` statement can be changed to an `if` statement, and the condition can be concatenated with the previous `if`.

Reply:
Changed in https://github.com/cryptocoinjs/secp256k1-node/commit/b963603f2c0f556f572e6d1ac72cb054e5fce78d